griffon/services/core_queries.py

```diff
@@ -194,32 +194,28 @@ def execute(self, status=None) -> dict:
         return c["product_streams"]
 
 
-def async_retrieve_sources(self, purl, max_results=5000):
-    source_params = {
-        # "max_results": max_results,
+def async_retrieve_sources(self, purl):
+    params = {
+        "limit": 300,
         "latest_components": "True",
         "root_components": "True",
         "provides": purl,
         "include_fields": "type,nvr,purl,name,namespace,download_url,related_url",
     }
-    return [
-        source_component.to_dict()
-        for source_component in self.components.retrieve_list_iterator_async(**source_params)
-    ]
```

Comment:
What is the purpose of removing the `to_dict`?  I mean we are working with the list of components as dict and then with its upstreams and sources we are using the internal bindings Component model, shouldn't we unify?

Reply:
want to centralise all post processing into output.py ... this might be surprising if this module gets imported outside of griffon usage (eg. service reuse) ... hoping that makes sense.